src/entry.rs

```diff
@@ -49,6 +49,16 @@ pub enum EntryIo<'a> {
     Data(io::Take<&'a ArchiveInner<Read + 'a>>),
 }
 
+/// When unpacking items the unpacked thing is returned to allow custom
+///  additional handling by users.
+#[derive(Debug)]
+pub enum Unpacked {
+    /// A file was unpacked.
+    File(std::fs::File),
+    /// A directory, hardlink, symlink, or other node was unpacked.
+    Other,
```

Comment:
Could this perhaps be a nonexhaustive enum? Basically rename this to `__Nonexhaustive` and put `#[doc(hidden)]` on it with other docs indicating that it may be extended in the future. It seems like we'd probably want to extend this in the future to indicate directories that were opened, hard links created, etc.

Reply:
I think I've done this right.